packages/flutter/test/widgets/overlay_portal_test.dart

```diff
@@ -2247,6 +2246,34 @@ void main() {
       expect(overlayRenderObject, isNot(paints..paragraph()..paragraph()));
     });
   });
+
+  testWidgets('Using OverlayPortal with no child in ListView', (WidgetTester tester) async {
+    final bool debugVisitOffstageChildren = Element.debugVisitOffstageChildren;
+    Element.debugVisitOffstageChildren = true;
+    addTearDown(() {
+      Element.debugVisitOffstageChildren = debugVisitOffstageChildren;
+    });
```

Comment:
This is strange that you only get the correct behavior when you set some kind of a secret flag.